roles/contiv_network/tasks/v2plugin.yml

```diff
@@ -1,32 +1,46 @@
 ---
 # This role contains tasks for starting docker plugin service
 
+- name: for v2 plugin, extract netctl binaries
+  shell: tar vxjf {{ contiv_network_dest_file }} netctl contrib/completion/bash/netctl
+  args:
+    chdir: /usr/bin/contiv/netplugin
+
 - name: create the directories needed for ovs
-  shell: mkdir -p {{ item }}
+  file: state=directory path={{ item }}
   with_items:
     - "/etc/openvswitch"
     - "/var/log/openvswitch"
 
 - name: check if v2plugin has been installed on {{ run_as }} nodes
   shell: docker plugin ls | grep -n {{ contiv_v2plugin_image }}
   register: v2plugin_installed
+  changed_when: no
   ignore_errors: True
 
-- name: install v2plugin on {{ run_as }} nodes
+- name: check if v2plugin archive is in the contiv_cache
+  local_action: stat path=/var/contiv_cache/{{ contiv_v2plugin_tar_filename }} follow=yes
```

Comment:
why not use `stat` module and make it a local operation ?

Reply:
local_action is a local operation using the stat module, it's a shorthand for delegate_to: 127.0.0.1:
```http://docs.ansible.com/ansible/latest/playbooks_delegation.html#delegation```